src/lib/rest/StringFilter.cpp

```diff
@@ -976,32 +1225,59 @@ bool StringFilter::mongoFilterPopulate(std::string* errorStringP)
     BSONObjBuilder     bb;
     BSONObjBuilder     bb2;
     BSONObj            f;
+    std::string        left = std::string(itemP->left.c_str());
+
+    //
+    // Left hand side might have to change, in case of Metadata filters (mq)
+    // The change consists in adding a '.md.' between attribute-name and metadata-name.
+    // This is to make the queries in mongo work.
+    //
+    if (type == SftMq)
+    {
+      if (itemP->attributeName == "")
+      {
+        *errorStringP = "no attribute name - not valid for metadata filters";
+        return false;
+      }
+
+      if (itemP->metadataName == "")
+      {
+        *errorStringP = "no metadata name - not valid for metadata filters";
+        return false;
+      }
 
-    if (itemP->left == DATE_CREATED)
+      left = itemP->attributeName + ".md." + itemP->metadataName;
```

Comment:
Sure, however, CSUB_EXPR_MQ is not the right constant.
We need a constant for name of the vector of the metadata used in mongo. 
I guess ENT_ATTRS_MD is what should be used ...